libraries/cyclestreets-core/src/main/java/net/cyclestreets/api/client/JourneyStringTransformer.kt

```diff
@@ -0,0 +1,68 @@
+package net.cyclestreets.api.client
+
+import com.bazaarvoice.jolt.Chainr
+import com.bazaarvoice.jolt.JsonUtils
+
+import fr.arnaudguyon.xmltojsonlib.XmlToJson
+
+object JourneyStringTransformer {
+
+    fun fromV1ApiXml(inputXml: String): String {
+        val xmlToJson = XmlToJson.Builder(inputXml).build()
+        val intermediateJson = xmlToJson.toString()
+        return joltTransform(intermediateJson, V1API_XML_JOLT_SPEC)
+    }
+
+    fun fromV1ApiJson(inputJson: String): String {
+        return joltTransform(inputJson, V1API_JSON_JOLT_SPEC)
+    }
+
+    // Uses https://github.com/bazaarvoice/jolt to transform JSON strings, without
+    // the need for us to define intermediate object representations.
+    private fun joltTransform(inputJson: String, specJson: String): String {
+        val inputJsonObject = JsonUtils.jsonToObject(inputJson)
+
+        val chainrSpecJson = JsonUtils.jsonToList(specJson)
+        val chainr = Chainr.fromSpec(chainrSpecJson)
+
+        val transformedOutput = chainr.transform(inputJsonObject)
+        return JsonUtils.toJsonString(transformedOutput)
+    }
+
+    private const val V1API_XML_JOLT_SPEC = """[{
+  "operation": "shift",
+  "spec": {
+    "markers": {
```

Comment:
Multiline strings are one nicety we're taking advantage of by using Kotlin here.

Reply:
Which is a fairly large win here :)